SyncExtension/UseCases/Files/CreateFileUseCase.swift

```diff
@@ -0,0 +1,112 @@
+//
+//  CreateFileUseCase.swift
+//  SyncExtension
+//
+//  Created by Robert Garcia on 10/8/23.
+//
+
+import Foundation
+import FileProvider
+import InternxtSwiftCore
+import os.log
+
+enum CreateFileUseCaseError: Error {
+    case InvalidParentId
+    case CannotOpenInputStream
+    case MissingDocumentSize
+}
+
+
+struct CreateFileUseCase {
+    let logger = Logger(subsystem: "com.internxt", category: "CreateFile")
+    private let cryptoUtils = CryptoUtils()
+    private let encrypt: Encrypt = Encrypt()
+    private let trashAPI: TrashAPI = APIFactory.Trash
+    private let item: NSFileProviderItem
+    private let encryptedFileDestination: URL
+    private let fileContent: URL
+    private let networkFacade: NetworkFacade
+    private let completionHandler: (NSFileProviderItem?, NSFileProviderItemFields, Bool, Error?) -> Void
+    private let driveAPI = APIFactory.Drive
+    private let config = ConfigLoader().get()
+    init(item: NSFileProviderItem, url: URL, encryptedFileDestination: URL, completionHandler: @escaping (NSFileProviderItem?, NSFileProviderItemFields, Bool, Error?) -> Void) {
+        self.item = item
+        
+        self.fileContent = url
+        self.encryptedFileDestination = encryptedFileDestination
+        self.completionHandler = completionHandler
+        self.networkFacade = NetworkFacade(mnemonic: config.MNEMONIC, networkAPI: APIFactory.Network)
```

Comment:
guard this mnemonic in some point to ensure that it has the proper form (n random words) to prevent content corruption on encryption / decryption

Reply:
I'll do this once I implement the auth system totally, some values like the tokens, mnemonic and user data will be saved in the device, this one is coming from a JSON just to act like im logged in